### ISSUE TITLE
Demo outputs colour-shifted image without inference result on detection model

### ISSUE THREAD

**Issue by QSmally** [open-edge-platform/training_extensions#2062]
**Describe the bug**

I'm running a generated model through the demo, however, its output image is colour-shifted and there are no inference results rendered on it. It runs correctly on an Luxonis OAK pipeline when converted to a MyriadX/blob format.

**Steps to Reproduce**

1. Have a `demo` output from `otx deploy`;
2. Run the below command (Dockerfile given at the end);
3. See `.../images/outputs/someimage.png` at the host path.

```docker
$ docker run \
    --rm \
    -v "/path/to/model:/mnt/model:ro" \
    -v "/path/to/images:/mnt/images:rw" \
    modeldemo \
    python /demo/demo.py -i /mnt/images/someimage.png -m /mnt/model --output /mnt/images/outputs
```

**Environment:**

- OS: `python:3.8` (expands to `debian:11`) 
- Framework version: `openvinotoolkit/training_extensions` commit 2a05810 
- Python version: `python3.8`
- OpenVINO version: `openvino 2022.3.0`
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Input and output**:

<img width="1051" alt="othercar" src="https://user-images.githubusercontent.com/38661077/233982119-89d8f9ce-c592-4cf1-b1fe-95863bb72844.png">

![othercar](https://user-images.githubusercontent.com/38661077/233982089-acf72656-479b-4d06-842e-3b11a6eb3209.png)

**Dockerfile**

```dockerfile
FROM python:3.8

RUN apt update && apt install ffmpeg libsm6 libxext6 libgl1-mesa-glx -y

WORKDIR /demo
COPY . /demo
ENV PYTHONPATH=$PYTHONPATH:/demo/model_wrappers

RUN python -m pip install -r requirements.txt && \
    python -m pip install torch==1.13.1 torchvision==0.14.1 torchaudio==0.13.1 --extra-index-url https://download.pytorch.org/whl/cpu
```

**config.json**

```json
{
    "type_of_model": "OTX_SSD",
    "converter_type": "DETECTION",
    "model_parameters": {
        "result_based_confidence_threshold": true,
        "confidence_threshold": 0.15000000596046448,
        "labels": {
            "label_tree": {
                "type": "tree",
                "directed": true,
                "nodes": [],
                "edges": []
            },
            "label_groups": [
                {
                    "_id": "6443d9946b4616aebe1b6518",
                    "name": "labels",
                    "label_ids": [
                        "1"
                    ],
                    "relation_type": "EXCLUSIVE"
                }
            ],
            "all_labels": {
                "1": {
                    "_id": "1",
                    "name": "plate",
                    "color": {
                        "red": 24,
                        "green": 121,
                        "blue": 122,
                        "alpha": 255
                    },
                    "hotkey": "",
                    "domain": "DETECTION",
                    "creation_date": "2023-04-22T12:56:51.810000",
                    "is_empty": false,
                    "is_anomalous": false
                }
            }
        }
    },
    "tiling_parameters": {
        "visible_in_ui": true,
        "type": "PARAMETER_GROUP",
        "enable_tiling": false,
        "enable_tile_classifier": false,
        "enable_adaptive_params": true,
        "tile_size": 400,
        "tile_overlap": 0.2,
        "tile_max_number": 1500,
        "header": "Tiling Parameters",
        "description": "Tiling Parameters",
        "_ParameterGroup__metadata_overrides": {},
        "groups": [],
        "parameters": [
            "enable_adaptive_params",
            "enable_tile_classifier",
            "enable_tiling",
            "tile_max_number",
            "tile_overlap",
            "tile_size"
        ]
    }
}
```

**Comment by QSmally** [comment]
Ty for the reference for the colour shift, @goodsong81

It doesn't address the un-rendered predictions, though? `otx explain` outputs the mapping correctly and prints that there have been 41 predictions made out of 43 images.

<img width="221" alt="image" src="https://user-images.githubusercontent.com/38661077/234018752-cf99cc83-8a7a-4dc0-ab8b-d1737a679f85.png">

How can I overlay the map with the original image in the explanation?

**Comment by GalyaZalesskaya** [comment]
Thank you so much for reporting this issue. It was fixed in [PR#2051](https://github.com/openvinotoolkit/training_extensions/pull/2051/)
